crispy_bootstrap5/bootstrap5.py

```diff
@@ -7,3 +12,130 @@ class FloatingField(Field):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.attrs["placeholder"] = self.fields[0]
+
+
+class Container(Div):
+    """
+    Base class used for `Tab` and `AccordionGroup`, represents a basic container concept
+    """
+
+    css_class = ""
+
+    def __init__(self, name, *fields, **kwargs):
+        super().__init__(*fields, **kwargs)
+        self.template = kwargs.pop("template", self.template)
+        self.name = name
+        self._active_originally_included = "active" in kwargs
+        self.active = kwargs.pop("active", False)
+        self.always_open = kwargs.pop("always_open", False)
+        if not self.css_id:
+            self.css_id = slugify(self.name)
+
+    def __contains__(self, field_name):
+        """
+        check if field_name is contained within tab.
+        """
+        return field_name in map(lambda pointer: pointer[1], self.get_field_names())
+
+    def render(self, form, form_style, context, template_pack=TEMPLATE_PACK, **kwargs):
+        if self.active:
+            if "active" not in self.css_class:
+                self.css_class += " active"
+        else:
+            self.css_class = self.css_class.replace("active", "")
+        return super().render(form, form_style, context, template_pack)
+
+
+class ContainerHolder(Div):
+    """
+    Base class used for `TabHolder` and `Accordion`, groups containers
+    """
+
+    def first_container_with_errors(self, errors):
+        """
+        Returns the first container with errors, otherwise returns None.
+        """
+        for tab in self.fields:
+            errors_here = any(error in tab for error in errors)
+            if errors_here:
+                return tab
+        return None
+
+    def open_target_group_for_form(self, form):
+        """
+        Makes sure that the first group that should be open is open.
+        This is either the first group with errors or the first group
+        in the container, unless that first group was originally set to
+        active=False.
+        """
+        target = self.first_container_with_errors(form.errors.keys())
+        if target is None:
+            target = self.fields[0]
+            if not getattr(target, "_active_originally_included", None):
+                target.active = True
+            return target
+
+        target.active = True
+        return target
+
+
+class AccordionGroup(Container):
+    """
+    Accordion Group (pane) object. It wraps given fields inside an accordion
+    tab. It takes accordion tab name as first argument::
+
+        AccordionGroup(
+            "group name",
+            "form_field_1",
+            "form_field_2",
+            active=True,
+            always_open=True
+            )
+    """
+
+    template = "%s/accordion-group.html"
+    data_parent = ""  # accordion parent div id.
+
+
+class Accordion(ContainerHolder):
```

Comment:
Can we inherit from accordion in the core package and then make the adjustments we need here? I think this will avoid copying the other classes quite so verbatim? 

I guess we should then rename this class to something like BS5Accordion. 🤔

Reply:
Yeah, this was the point I was most unsure about. I even thought that maybe bootstrap5 should be treated the same way as other template packs, ie tailwind, and pull all the Classes from the core... This way all the layout classes would be imported from the template pack, not from the core.

Creating a new class like BS5Accordion seems reasonable enough for me, my only fear is for future users not knowing which class to use for their layouts. I definetly don't have a clear vision on this subject.

Thanks for all the comments, by the way! I'll wait for your replies and when you decide what's best I'll make the due changes.

Reply:
I _think_ the story should be something like.

- you want to upgrade? Just install the new package and change your crispy template setting to `bootstrap5`
- hey, did you know there's some new features? Use the new layout classes for those.

We'd therefore leave the existing classes in core and add the new classes here?

Reply:
I think you're right. That seems intuitive enough for me.
We just have to make sure the use cases I'm tweaking are on the readme, like you suggested, so future users can use those features hassle free.

I'll modify the classes then

Reply:
I created the classes BS5Accordion and BS5AccordionGroup.
I also changed the accordion tests to use those new classes.
I think I should probably make more tests, perhaps? Since we're keeping Accordion and AccordionGroup from the core pack, we should probably keep tests with those classes as well?

Reply:
> we should probably keep tests with those classes as well?

Sounds reasonable.

Reply:
Done! Sorry for the excessive amount of commits and thank you for taking the time to review this